### PR TITLE
[Fabric] Add caretHidden to TextInput

### PR DESCRIPTION
**Pull request by TatianaKapos**
## Description
Adds caretHidden to Fabric's TextInput

### Type of Change
- New feature (non-breaking change which adds functionality)

### Why
Component Parity with Paper

### What
Overrides RichEdit controlling the visibility of the caret visual if caretHidden is set to true.

## Screenshots
https://github.com/microsoft/react-native-windows/assets/42554868/085cfe07-1837-4f5e-8be2-3a64ba1811f1

## Testing
tested locally

## Changelog
no
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12430)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp [hunk 1]
```diff
@@ -161,7 +161,7 @@ struct CompTextHost : public winrt::implements<CompTextHost, ITextHost> {
 
   //@cmember Show the caret
   BOOL TxShowCaret(BOOL fShow) override {
-    m_outer->ShowCaret(fShow);
+    m_outer->m_props->caretHidden ? m_outer->ShowCaret(false) : m_outer->ShowCaret(fShow);
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp:164]: To me this reads more clearly:
```suggestion
    m_outer->ShowCaret(m_outer->m_props->caretHidden != TRUE);
```
As in the action is on the left, and the conditional operator is only on the arguments.
Part of what's happening here is a conversion from `BOOL` to `bool`. Which is... ugh, legacy. `BOOL` is an integer, not a true boolean. Some places there's a `BOOLIFY` macro, but I don't know if we have it (if so, use it). Other places people will use `!!value` to force it from `BOOL` to `bool`. Another form is to explicitly check for the defined `TRUE` value. Anyway, mostly my feedback was on the conditional operator; the `BOOL` stuff should match other places i nthe source.

File: vnext/Microsoft.ReactNative/Fabric/Composition/CompositionContextHelper.cpp [hunk 2]
```diff
@@ -1043,6 +1043,7 @@ struct CompCaretVisual
     // TODO should make the caret use an invert brush by default
     m_compVisual.Brush(m_compositor.CreateColorBrush({255, 0, 0, 0} /* Black */));
     m_compVisual.Opacity(1.0f);
+    m_isVisible = true;
```
Comment (chrisglein) [inline vnext/Microsoft.ReactNative/Fabric/Composition/CompositionContextHelper.cpp:1046]: On line 1098 this is initialized to false. Should that be updated instead?

Comment (TatianaKapos) [reply vnext/Microsoft.ReactNative/Fabric/Composition/CompositionContextHelper.cpp]: ahhh good catch! That makes more sense than updating it here :)

File: vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp (hunk 1 above)
Comment (TatianaKapos) [reply vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp:164]: Ah good to know! I actually didn't know about the `BOOL` operator but have definitely seen it before and assumed it was a regular `bool`. 

I think in Fabric caretHidden is defined [as a regular `bool`](https://github.com/microsoft/react-native-windows/blob/main/vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputProps.h#L116), so there isn't a conversion happening in this file. Also, maybe I'm reading this wrong, but how does the suggested change use `fShow` if `caretHidden` is false? Or did you mean `m_outer->ShowCaret(m_outer->m_props->caretHidden != TRUE? false : fShow);`?

Comment (chrisglein) [reply vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp:164]: Oh, I left out `fShow`, didn't I? Turns out suggestions are just that... suggestions. Yes, something more like this:
```suggestion
    m_outer->ShowCaret(m_outer->m_props->caretHidden ? false : fShow);
```

**Review by jonthysell — APPROVED** [review]

**Comment by chiaramooney** [comment]
Could you also add a test to TextInputComponentTest.test.ts and call DumpVisualTree on the testID you added? Example here! https://github.com/microsoft/react-native-windows/blob/1582219c0f9efa529bbb68e4ae1b68cc9c4f75e4/packages/e2e-test-app-fabric/test/TextInputComponentTest.test.ts#L638-L642

**Review by chiaramooney — APPROVED** [review]
👍

**Pull request merged**